kolibri/core/content/utils/content_request.py

```diff
@@ -367,8 +402,19 @@ def _process_content_requests(incomplete_downloads):
 
     # loop while we have pending downloads
     while incomplete_downloads_with_metadata.exists():
+        # if a limit is set, subtract the total content storage size from the limit to get free space
+        if get_device_setting("set_limit_for_autodownload", False):
+            # compute total space used by automatic and learner initiated downloads
+            completed_downloads_size = _total_size(completed_downloads_queryset())
+            # convert limit_for_autodownload from GB to bytes
+            auto_download_limit = bytes_from_humans(
+                str(get_device_setting("limit_for_autodownload", "0")) + "GB"
+            )
+            free_space = auto_download_limit - completed_downloads_size
+        else:
+            free_space = get_free_space(OPTIONS["Paths"]["CONTENT_DIR"])
```

Comment:
I think we should always set the `free_space` to this before the `if` statement here.

Then in the `if` statement, we should do `free_space = min(free_space, auto_download_limit - completed_downloads_size)` - as we don't want the reported free space to exceed the space available on disk!